src/rngs/adapter/reseeding.rs

```diff
@@ -16,52 +16,67 @@ use core::mem::size_of;
 use rand_core::{RngCore, CryptoRng, SeedableRng, Error, ErrorKind};
 use rand_core::block::{BlockRngCore, BlockRng};
 
-#[cfg(all(feature="std", unix, not(target_os="emscripten")))]
-extern crate libc;
-
-#[cfg(all(feature="std", unix, not(target_os="emscripten")))]
-use std::sync::atomic::{AtomicUsize, ATOMIC_USIZE_INIT,
-                        AtomicBool, ATOMIC_BOOL_INIT, Ordering};
-
-
-/// A wrapper around any PRNG which reseeds the underlying PRNG after it has
-/// generated a certain number of random bytes.
+/// A wrapper around any PRNG that implements [`BlockRngCore`], that adds the
+/// ability to reseed it.
 ///
-/// When the RNG gets cloned, the clone is reseeded on first use.
+/// `ReseedingRng` reseeds the underlying PRNG in the following cases:
 ///
-/// Reseeding is never strictly *necessary*. Cryptographic PRNGs don't have a
-/// limited number of bytes they can output, or at least not a limit reachable
-/// in any practical way. There is no such thing as 'running out of entropy'.
+/// - On a manual call to [`reseed()`].
+/// - After `clone()`, the clone will be reseeded on first use.
+/// - After a process is forked, the RNG in the child process is reseeded on
+///   first use. Several values may be returned before the fork is detected.
```

Comment:
After a process is forked, the RNG in the child process is reseeded *within the next few generated values, depending on the block size of the underlying PRNG. For `ChaChaRng` and `Hc128Rng` this is a maximum of 15 `u32` values before reseeding.*

Your doc isn't quite accurate, and also not very precise.

Reply:
Yes, better :+1: